tests/DelayStampTestTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the zenstruck/messenger-test package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Messenger\Test\Tests;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\Clock\Test\ClockSensitiveTrait;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Stamp\DelayStamp;
+use Zenstruck\Messenger\Test\InteractsWithMessenger;
+use Zenstruck\Messenger\Test\Tests\Fixture\Messenger\MessageA;
+use Zenstruck\Messenger\Test\Tests\Fixture\Messenger\MessageB;
+use Zenstruck\Messenger\Test\Tests\Fixture\Messenger\MessageC;
+
+/**
+ * @author Nicolas PHILIPPE <nikophil@gmail.com>
+ */
+final class DelayStampTestTest extends WebTestCase
```

Comment:
one thing I did not managed to do is to combine `support_delay_stamp` with unblocking behavior.
Do you think we can manage to have them both work?
ie, this kind of code could work:
```php
        $clock = self::mockTime();

        $transport = $this->transport('async')->unblock();
        
        $transport->send(new Envelope(new MessageB()));
        $transport->send(new Envelope(new MessageC(), [new DelayStamp(5_000)]));

        $transport->acknowledged()->assertCount(0);

        $transport->process();
        
        $transport->acknowledged()->assertCount(1)->assertContains(MessageB::class);

        $clock->sleep(5);
        $transport->acknowledged()->assertCount(2)->assertContains(MessageC::class);
```

or if this is impossible to make it work (I think it is), maybe we can say something about it in the docs?

Reply:
I feel like specifically unblocking should set `support_delay_stamp=false` but not sure... What if you had `test://?intercept=false&support_delay_stamp=true`? Exception?

Reply:
not sure what to do about this :thinking: 

at first I was thinking `support_delay_stamp=true` should be the default, and always activated (hence the deprecation) but this would break unblocking behavior

> I feel like specifically unblocking should set `support_delay_stamp=false` but not sure

I have the same feeling, but I don't like to transparently disable the behavior

Reply:
on the other hand.... both feature are not TOTALLY incompatible

the unique problem would be "delayed" message cannot be handled by the unblocking mechanism, a `process()` (or a `dispatch()` of another message) must be called after a `sleep()` has been called, that's all.

we have two options:
- do nothing about that, and users, should call `process()` after a `seep()`, in order to handle delayed messages
- automatically deactivate the support of delay stamp as soon as intercept is deactivated.  And `test://?intercept=false&support_delay_stamp=true` would not throw an exception because not supporting delay stamp is not a valid configuration (and the option will be dropped in a further version)

either we chose one or the other solution, we must advertise it in the docs

WDYT?

by the way, I can't see any solution to have both configurations work nicely together. A solution would have been to decorate the clock, in order to process after any call to `sleep()` but the sleep method is part of symfon's contract, and not PSR's clock. Another solution would be to create a hard dep on `symfony/clock`. This is a solution we could consider

Reply:
> do nothing about that, and users, should call process() after a seep(), in order to handle delayed messages

Let's do this. Seems easier to reason about to me.

Reply:
agreed, I've added some warning about this case in the docs